src/search/FindInFiles.js

```diff
@@ -734,67 +742,81 @@ define(function (require, exports, module) {
          * Remove existing search results that match the given entry's path
          * @param {(File|Directory)} entry
          */
-        function _removeSearchResultsForEntry(entry) {
-            Object.keys(searchModel.results).forEach(function (fullPath) {
-                if (fullPath === entry.fullPath ||
-                        (entry.isDirectory && fullPath.indexOf(entry.fullPath) === 0)) {
-                    // node search : inform node that the file is removed
-                    filesRemoved([fullPath]);
-                    if (findOrReplaceInProgress) {
-                        searchModel.removeResults(fullPath);
-                        resultsChanged = true;
+        function _removeSearchResultsForEntries(entries) {
+            var fullPaths = [];
+            entries.forEach(function (entry) {
```

Comment:
Might be worth checking out if couple of `Array.prototype.map`s would be faster than nested `forEach`s and a push